common/arduino101_services/arduino101_main.c

```diff
@@ -26,21 +26,73 @@
 #define RESET_STACKSIZE     384
 #define USBSERIAL_STACKSIZE 384
 
-
 /* scheduling priority used by each thread */
 #define PRIORITY 7
 #define TASK_PRIORITY 8
 
+#define ARDUINO_101_VID		0x8087
+#define ARDUINO_101_PID		0x0AB6
+
+#define MAX_DESCRIPTOR_LENGTH	64
+
 char __noinit __stack sketch_stack_area[SKETCH_STACKSIZE];
 char __noinit __stack cdcacm_setup_stack_area[CDCACM_STACKSIZE];
 char __noinit __stack baudrate_reset_stack_area[RESET_STACKSIZE];
 char __noinit __stack usb_serial_stack_area[USBSERIAL_STACKSIZE];
 
+const char *vendor = "Intel";
+const char *product =  "Arduino 101";
+const char *serial = "00.01";
+
+struct gpio_callback cb;
+
+void copy_device_descriptor(char *dest, uint8_t *source, const char *default_value, uint8_t len, uint8_t max)
+{
+	if ((len > 0) && (len < max))
+	{
+		memcpy(dest, source, len);
+		dest[len] = '\0';
+	}
+	else
+	{
+		memcpy(dest, default_value, strlen(default_value));
```

Comment:
So if the passed string is too long, it uses default values instead? That's a bit weird. I think most people would expect their long string to be truncated to fit, instead.